File: Day 2/Trim a Binary Search Tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode trimBST(TreeNode root, int low, int high) {
        root=trimBSTutil(root,low,high);
        return root;
    }
    public TreeNode trimBSTutil(TreeNode root,int low,int high){
        if(root==null){
            return null;
        }
        
        if(root.val<=high && root.val>=low){
            root.left=trimBSTutil(root.left,low,high);
            root.right=trimBSTutil(root.right,low,high);
        }
        else{
            if(root.val>high){
                System.out.println("Left");
                root= trimBSTutil(root.left,low,high);
            }
            else{
                System.out.println("Left");
                root= trimBSTutil(root.right,low,high);
            }
        }
        return root;
    }
}

File: Day 2/Trim a Binary Search Tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def trimBST(self, root, low, high):
        """
        :type root: TreeNode
        :type low: int
        :type high: int
        :rtype: TreeNode
        """
        if root==None:
            return None
        if root.val>=low and root.val<=high:
            root.left=self.trimBST(root.left,low,high)
            root.right=self.trimBST(root.right,low,high)
        else:
            if root.val>high:
                root=self.trimBST(root.left,low,high)
            else:
                root=self.trimBST(root.right,low,high)
        return root
        
